pkg/storegateway/gateway_test.go

```diff
@@ -655,9 +655,9 @@ func TestStoreGateway_SyncOnRingTopologyChanged(t *testing.T) {
 				desc.AddIngester("instance-2", "127.0.0.2", "", ring.Tokens{4, 5, 6}, ring.JOINING, registeredAt)
 			},
 			updateRing: func(desc *ring.Desc) {
-				instance := desc.Ingesters["instance-2"]
+				instance := desc.Ingesters["instance-1"]
 				instance.Addr = "127.0.0.3"
-				desc.Ingesters["instance-2"] = instance
+				desc.Ingesters["instance-1"] = instance
```

Comment:
This is the test case that was missing, where new instance address will change its order in list of addresses from the ring (but instance ID is preserved). For store gateway, we do not want ring to sync in this scenario as long as tokens are persisted.